sqlmesh/utils/cron.py

```diff
@@ -21,12 +23,37 @@ def cron_prev(cron: str, time: TimeLike) -> float:
 class CroniterCache:
     def __init__(self, cron: str, time: t.Optional[TimeLike] = None):
         self.cron = cron
-        self.curr: TimeLike = time or now()
-
-    def get_next(self) -> float:
-        self.curr = cron_next(self.cron, self.curr)
+        self.curr: TimeLike = now() if time is None else time
+        self._interval_seconds: t.Optional[int] = None
+
+    @property
+    def interval_seconds(self) -> int:
+        if self._interval_seconds is None:
+            seconds = set()
+            curr = to_datetime(self.curr)
+
+            for _ in range(10):
+                prev = curr
+                curr = to_datetime(cron_next(self.cron, curr))
+                seconds.add(curr - prev)
+
+            if len(seconds) == 1:
+                self._interval_seconds = first(seconds).seconds
+            else:
+                self._interval_seconds = 0
```

Comment:
Why 0?

Reply:
Because it's not `None` but still behaves as false? I'd prefer this to be more explicit

Reply:
docstring